WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -199,7 +203,8 @@ def create
       redirect_to competition_path(@competition)
       return
     end
-    @registration = @competition.registrations.build(registration_params.merge(user_id: current_user.id))
+    @registration = Registration.find_by_user_id_and_competition_id(current_user.id, @competition.id) || @competition.registrations.build(user_id: current_user.id)
```

Comment:
What if the Registration exists but is not marked as `deleted`? Would this go on to mutate it?